static/js/shared-stories-filtering.js

```diff
@@ -0,0 +1,27 @@
+let abuse = document.getElementById("abuse-checkbox");
+let drug = document.getElementById("drug-checkbox");
+let negbody = document.getElementById("negbody-checkbox");
+let violence = document.getElementById("violence-checkbox");
+let mentalhealth = document.getElementById("mentalhealth-checkbox");
+let other = document.getElementById("other-checkbox");
+let all = document.getElementById("all-checkbox");
+
+let searchForm = document.getElementById("search-form");
+
+abuse.addEventListener("change", pressSubmit);
+drug.addEventListener("change", pressSubmit);
+negbody.addEventListener("change", pressSubmit);
+violence.addEventListener("change", pressSubmit);
+mentalhealth.addEventListener("change", pressSubmit);
+other.addEventListener("change", pressSubmit);
+all.addEventListener("change", isAllChecked)
+
+function isAllChecked(){
+    if(all.checked){
+        pressSubmit();
+    }
+}
+
+function pressSubmit() {
+    searchForm.submit();
+}
```

Comment:
One nice way (I think) to simplify this would be to use jQuery's ability to act on all elements with the same class simultaneously. That way you could reduce your code to something like this:
```javascript
$(function() {
  $(".custom-control-input").on("change", pressSubmit)

  function pressSubmit() {
      $("#search-form").submit()
  }
})
```
This skips the `isAllChecked` test, because I think in practice that will never apply (because the `all` checkbox is always unchecked when the page is loaded).